systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -94,297 +105,147 @@ public static HelmClient helmClient() {
-                .filter(p -> p.getMetadata().getName().startsWith(KafkaMirrorMaker2Resources.deploymentName(kafkaMirrorMaker2.getMetadata().getName())))
-                .forEach(p -> PodUtils.deletePodWithWait(p.getMetadata().getName()));
+    public final <T extends HasMetadata> boolean waitResourceCondition(T resource, Predicate<T> condition, TimeoutBudget timeout) {
+        assertNotNull(resource);
+        assertNotNull(resource.getMetadata());
+        assertNotNull(resource.getMetadata().getName());
+        ResourceType<T> type = findResourceType(resource);
+        assertNotNull(type);
+
+        while (!timeout.timeoutExpired()) {
+            T res = type.get(resource.getMetadata().getNamespace(), resource.getMetadata().getName());
+            if (condition.test(res)) {
+                LOGGER.info("Resource {} in namespace {} is {}}!", resource.getMetadata().getName(), resource.getMetadata().getNamespace(), condition.toString());
+                return true;
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                LOGGER.info("Resource {} in namespace {} is not {}!", resource.getMetadata().getName(), resource.getMetadata().getNamespace(), condition.toString());
+                Thread.currentThread().interrupt();
+                return false;
+            }
+        }
+        T res = type.get(resource.getMetadata().getNamespace(), resource.getMetadata().getName());
+        boolean pass = condition.test(res);
+        if (!pass) {
+            LOGGER.info("Resource {} with with type {} failed condition check: {}", resource.getMetadata().getName(), type, resourceToString(res));
+        }
+        return pass;
     }
 
-    private static void waitForDeletion(KafkaBridge kafkaBridge) {
-        LOGGER.info("Waiting when all the Pods are terminated for KafkaBridge {}", kafkaBridge.getMetadata().getName());
-
-        DeploymentUtils.waitForDeploymentDeletion(KafkaBridgeResources.deploymentName(kafkaBridge.getMetadata().getName()));
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaBridgeResources.deploymentName(kafkaBridge.getMetadata().getName()));
+    private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory());
 
-        kubeClient().listPods().stream()
-                .filter(p -> p.getMetadata().getName().startsWith(KafkaBridgeResources.deploymentName(kafkaBridge.getMetadata().getName())))
-                .forEach(p -> PodUtils.deletePodWithWait(p.getMetadata().getName()));
+    public static <T extends HasMetadata> String resourceToString(T resource) {
+        if (resource == null) {
+            return "null";
+        }
+        try {
+            return MAPPER.writeValueAsString(resource);
+        } catch (JsonProcessingException e) {
+            LOGGER.error("Resource {} failed converting resource to YAML: {}", resource.getMetadata().getName(), e.getMessage());
+            return "unknown";
```

Comment:
Wouldn't be better to return null or raise something? Just asking

Reply:
I think in this case it doesn't matter much.. :)